keys-component.js

```diff
@@ -84,26 +95,19 @@ function factory($scope, $routeParams, brAlertService, brKeyService,
     }).then(function() {
       $scope.$apply();
     });
-  };
+  }
+
+  // FIXME" what is current convention for this?
+  init(self.identity);
 
-  $scope.$watch('identity', function(value) {
+  $scope.$watch('self.identity', function(value) {
```

Comment:
New convention is:

```js
self.$onChanges = function(changes) {
  if(changes.identity) {
    // do something with `changes.identity.currentValue`
  }
};
```

But, as an aside, this `$watch` would not work properly here anyway, because evaluating the expression `'self.identity'` will always be undefined in the scope. If we weren't replacing the `$watch` with `$onChanges`, it would be:

```js
$scope.$watch(function() {
  return self.identity;
}, ...`

You could also do `'$ctrl.identity'` but that requires expression parsing and other things that are unnecessary and may potentially change; better to be explicit with a function when using `$watch` if you can. That being said, I think `$onChanges` will work better here.